internal/resolution/manifest/maven.go

```diff
@@ -490,7 +490,8 @@ func originalDependency(patch DependencyPatch, origDeps []DependencyWithOrigin)
 	dependency.ArtifactID = maven.String(IDs[1])
 
 	for _, d := range origDeps {
-		if d.Key() == dependency.Key() {
+		if d.Key() == dependency.Key() && d.Version != "" {
```

Comment:
this should solve the current problem of not being able to update dependency without a version since we agreed we would like to update dependency management section for this case. @michaelkedar